### PR TITLE
[ADR] First ADRs about API and Product Options

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | -----
| Branch?         | api
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

The idea of the ADR is to explain some architecture decisions (which we make a lot during a new API implementation). Such documents would also allow us to iterate easier over the implementation and change some decisions, if needed, in the future.

### CODE REVIEW

File: adr/api/2020-02-27-Product-Options.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## Description
+
+### Problem to solve
+
+Cover with API all `ProductOption`s related functionality provided by the Admin panel. 
+
+### Context
+
+New Sylius API (based on API Platform), should cover in 100% or business value provided in Admin panel (and already covered
```
Comment (GSadee) [inline adr/api/2020-02-27-Product-Options.md]:
```suggestion
New Sylius API (based on API Platform), should cover in 100% business value provided in Admin panel (and already covered
```

File: adr/api/2020-02-27-Product-Options.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+### Decision and reasoning
+
+- **API coverage:** for the beginning, we focus on implementing only features already covered with Behat UI scenarios.
+  In the future, it would result in bigger reliability of Behat API tests and allows more straightforward iteration over missing features.
+  
+  - As an example - we don't provide `ProductOptionsValue`s deletion feature in API, as it's not yet provided and
+    covered with Behat UI scenario (even though there is a value in this feature - and should be implemented separately for both, UI and API).
+  
+- **Translations:** should always be embedded as the collection of objects. They're irrelevant outside of the main object
+  and does not provide any value alone.
```
Comment (GSadee) [inline adr/api/2020-02-27-Product-Options.md]:
```suggestion
  and do not provide any value alone.
```

File: adr/api/2020-02-27-Product-Options.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  
+- **Product option values:** should always be provided with their IRIs. They can exist outside of the `ProductOption` resource
+  (e.g. be [related with](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Product/Model/ProductVariant.php#L44) `ProductVariant`).
+  Moreover, values embedded together with their translations in the `ProductOption` that already has its translations would
+  result in a massive and unmanageable response.
+  
+- **Updates of objects:** are defined with the PUT method, to allow embedded objects (like translations) update.
```
Comment (GSadee) [inline adr/api/2020-02-27-Product-Options.md]:
```suggestion
- **Updates of objects:** are defined with the PUT method, to allow embedded objects (like translations) to update.
```
?

File: adr/api/2020-03-03-API-Translations.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+## Description
+
+### Problem to solve
+
+Provide unified way to manage translations for translatable entities in API. 
+
+### Possible solutions
+
+The translations from collection could be embedded as objects within a `ProductOption`/`ProductOptionValue` resource
```
Comment (lchrusciel) [inline adr/api/2020-03-03-API-Translations.md]: Should we consider placing an option for flatten a list of translations to the object itself (similar to https://medium.com/locastic/having-troubles-with-implementing-translations-in-apiplatform-7861294fb7c6)?

Comment (Zales0123) [reply adr/api/2020-03-03-API-Translations.md]: We should consider it, but for sure not for the first version 😄

**Comment by lchrusciel** [comment]
Thanks, Mateusz! :1st_place_medal:

**Comment by lchrusciel** [comment]
Part of #11250

**Pull request merged**